qiskit_nature/algorithms/initial_points/hf_initial_point.py

```diff
@@ -0,0 +1,203 @@
+
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+
+from qiskit_nature.exceptions import QiskitNatureError
+from qiskit_nature.properties.second_quantization.electronic import ElectronicEnergy
+from qiskit_nature.circuit.library import UCC
+from qiskit_nature.properties.second_quantization.second_quantized_property import (
+    GroupedSecondQuantizedProperty,
+)
+
+from .initial_point import InitialPoint
+
+logger = logging.getLogger(__name__)
+
+
+class HFInitialPoint(InitialPoint):
+    """Compute the Hartree-Fock (HF) initial point.
+
+    All excitations will have zero coefficient and energy correction as the all-zero initial point
+    corresponds to the HF initial point once the HF initial state is prepended to the circuit.
+
+    A HF initial point uses the excitation list generated by the UCC ansatz to ensure that the shape
+    of the initial point is correct.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._reference_energy: float = 0.0
+        self._excitation_list: list[tuple[tuple[int, ...], tuple[int, ...]]] | None = None
+        self._corrections: dict[str, tuple[float, float]] | None = None
+
+    def __len__(self) -> int:
+        """The length of the computed initial point array."""
+        if self._corrections is None:
+            self.compute()
+        return len(self.to_numpy_array())
+
+    @property
+    def grouped_property(self) -> GroupedSecondQuantizedProperty:
+        """The grouped property.
+
+        The grouped property is not required to compute the HF initial point.
+        """
+        return self._grouped_property
+
+    @grouped_property.setter
+    def grouped_property(self, grouped_property: GroupedSecondQuantizedProperty) -> None:
+        if not isinstance(grouped_property, GroupedSecondQuantizedProperty):
+            logger.warning(
+                "grouped_property is not of type GroupedSecondQuantizedProperty; "
+                "grouped_property and reference_energy will not be set."
+            )
+            return
+
+        electronic_energy: ElectronicEnergy | None = grouped_property.get_property(ElectronicEnergy)
+        if not isinstance(electronic_energy, ElectronicEnergy):
+            logger.warning(
+                "ElectronicEnergy not obtained from grouped_property; "
+                "grouped_property and reference_energy will not be set."
+            )
+            return
+
+        self._reference_energy = electronic_energy.reference_energy if not None else 0.0
+        self._grouped_property = grouped_property
+
+    @property
+    def ansatz(self) -> UCC:
+        """The UCC ansatz.
+
+        This is used to ensure that the :attr:`excitation_list` matches with the UCC ansatz that
+        will be used with the VQE algorithm.
+
+        Raises:
+            QiskitNatureError: If not set using a valid
+                :class:`~qiskit_nature.circuit.library.ansatzes.ucc.UCC` instance.
```

Comment:
I don't see this being raised here. EDIT: ah hold on. This is a docstring for the getter AND setter, right?

Reply:
Yeah, I think the setter docstring is ignored by Sphinx, so it acts for both.